src/components/list/list.tsx

```diff
@@ -68,20 +68,16 @@ export class List {
         this.handleAction = this.handleAction.bind(this);
     }
 
-    public componentDidLoad() {
-        this.mdcList = new MDCList(
-            this.element.shadowRoot.querySelector('.mdc-list')
-        );
-
-        this.handleType();
+    public connectedCallback() {
+        this.setup();
     }
 
-    public componentDidUnload() {
-        if (this.selectable) {
-            this.mdcList.unlisten(ACTION_EVENT, this.handleAction);
-        }
+    public disconnectedCallback() {
+        this.teardown();
+    }
 
-        this.mdcList.destroy();
+    public componentDidLoad() {
```

Comment:
I made a bunch of comments, but I guess this is the only real change request 🙂

Reply:
`componentDidLoad` is only called once, but `connectedCallback` is called every time the component is connected to the DOM. In this case, we will not do anything the first time `connectedCallback` runs since we check for the `.mdc-list` element, which will only exist after the first `componentDidLoad`

Reply:
Hmm… yeah, ok, I see 🙂

Yeah, as long as we hedge it with the check for `element`, we're good. Kind of annoying that the first `connectedCallback` is different from the others in that way. If that's part of the native spec, and `componentDidLoad` isn't, I wonder how it's supposed you'd do something like this when using the vanilla web components…

Although I guess the native spec is a work in progress still…?